unic/ucd/name/src/name.rs

```diff
@@ -9,38 +9,138 @@
 // except according to those terms.
 
 use core::fmt;
+use core::cmp::Ordering;
 
-#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
-pub struct Name {
-    pieces: &'static [&'static str],
+pub static PREFIX_HANGUL_SYLLABLE: &'static str = "HANGUL SYLLABLE ";
+pub static PREFIX_CJK_UNIFIED_IDEOGRAPH: &'static str = "CJK UNIFIED IDEOGRAPH-";
+pub static PREFIX_TANGUT_IDEOGRAPH: &'static str = "TANGUT IDEOGRAPH-";
+pub static PREFIX_NUSHU_CHARACTER: &'static str = "NUSHU CHARACTER-";
+pub static PREFIX_CJK_COMPATIBILITY_IDEOGRAPH: &'static str = "CJK COMPATIBILITY IDEOGRAPH-";
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
+pub enum Name {
```

Comment:
This looks more clean now! Thanks, @eyeplum, for the update.

If you like working more on this component, would be great to add some documentation here with links to the specification of NR1, etc.

Here are two examples:

https://github.com/behnam/rust-unic/blob/master/unic/ucd/age/src/age.rs#L16-L39

https://github.com/behnam/rust-unic/blob/master/unic/ucd/bidi/src/bidi_class.rs#L17-L23

Also, you can add
```rust
#![deny(missing_docs)]
```
to `src/lib.rs` of this component, to require documentation on all public items on compile time.

Reply:
Sure! I will be working on `NR1` soon and will add the documentations.
Thank you so much for your guidance!